### PR TITLE
Snapshot TaskRun results in acceptance tests

### PR DESCRIPTION
**Pull request by zregvart**
Adds the ability to snapshot TaskRun results in acceptance tests and adds it to the current Task acceptance tests.

Ref. https://issues.redhat.com/browse/HACBS-2295

### CODE REVIEW

File: features/__snapshots__/task_validate_image.snap [hunk 1]
```diff
@@ -259,3 +259,31 @@ success: true
...
+[Strict with failures:results - 1]
+{}
+---
+
+[Non strict with failures:results - 1]
+{"TEST_OUTPUT":"{\"timestamp\":\"${TIMESTAMP}\",\"namespace\":\"\",\"successes\":0,\"failures\":1,\"warnings\":0,\"result\":\"FAILURE\"}"}
+---
+
+[Strict with warnings:results - 1]
+{"TEST_OUTPUT":"{\"timestamp\":\"${TIMESTAMP}\",\"namespace\":\"\",\"successes\":1,\"failures\":0,\"warnings\":0,\"result\":\"SUCCESS\"}"}
```
Comment (lcarva) [inline features/__snapshots__/task_validate_image.snap:276]: Is this a bug on this test and the one below? The name implies there should be at least one warning.

**Review by lcarva — APPROVED** [review]

File: features/__snapshots__/task_validate_image.snap (hunk 1 above)
Comment (zregvart) [reply features/__snapshots__/task_validate_image.snap:276]: Most likely a bug in the test, will investigate

**Comment by zregvart** [comment]
Will investigate and undraft when I figure out why we don't show warnings in a test that tests for warnings

**Comment by zregvart** [comment]
Should be okay now, fixed the test cases with warnings

**Review by lcarva — APPROVED** [review]
Nice! Thanks for addressing the test case :pray:

**Pull request merged**